vms/platformvm/state/metadata_validator.go

```diff
@@ -60,7 +58,7 @@ func parseValidatorMetadata(bytes []byte, metadata *validatorMetadata) error {
 		// potential reward and uptime was stored but potential delegatee reward
 		// was not
 		tmp := preDelegateeRewardMetadata{}
-		if _, err := txs.Codec.Unmarshal(bytes, &tmp); err != nil {
+		if _, err := metadataCodec.Unmarshal(bytes, &tmp); err != nil {
```

Comment:
This is the main change in the PR: I need to marshal/unmarshal validator metadata with their own codec metadataCodec, to be free to change version without impacting txs.Codec